pkg/encoding/v1/encoding_test.go

```diff
@@ -319,6 +319,35 @@ containers:
 				},
 			},
 		},
+
+		{
+			"Providing valid label strings",
```

Comment:
You are testing whole service here.

You should add test testing `(lb *Labels) UnmarshalYAML` something like  `TestLabels_UnmarshalYAML` similar to `TestEnvVariable_UnmarshalYAML` and others.